Amesos2:  Complex enabled build failure

## Bug Report
@trilinos/amesos2 
CC
@rppawlo 

I've been hitting a compile error with Complex (double) enabled in my HIP builds with Amesos2. 

I pulled develop yesterday, and still have the issue.

### Description
This build enables quite alot with HIP and Complex - using Cray Libsci as the host side BLAS/Lapack.

<details>
<summary> FAILED: packages/amesos2/src/CMakeFiles/amesos2.dir/Amesos2_Lapack.cpp.o  </summary>

```
[2145/3739] Building CXX object packages/amesos2/src/CMakeFiles/amesos2.dir/Amesos2_Lapack.cpp.o
FAILED: packages/amesos2/src/CMakeFiles/amesos2.dir/Amesos2_Lapack.cpp.o 
/opt/rocm-5.4.3/llvm/bin/amdclang++  -I/tmp/jjellio/build/amd-5.4.3_prgenv-amd_rocm-5.4.3_mpich-8.1.24_pure-amd_hip_amd-inlall-NOfunc-gfx90a_serial_atdm_complex_opt-g_cxx17_static/trilinos -I/g/g20/jjellio/src/github/Trilinos -I/tmp/jjellio/build/amd-5.4.3_prgenv-amd_rocm-5.4.3_mpich-8.1.24_pure-amd_hip_amd-inlall-NOfunc-gfx90a_serial_atdm_complex_opt-g_cxx17_static/trilinos/packages/amesos2/src -I/g/g20/jjellio/src/github/Trilinos/packages/amesos2/src -I/g/g20/jjellio/src/github/Trilinos/packages/amesos2/src/KLU2/Include -I/g/g20/jjellio/src/github/Trilinos/packages/amesos2/src/KLU2/Source -I/g/g20/jjellio/src/github/Trilinos/packages/common/auxiliarySoftware/SuiteSparse/src/AMD/Include -I/g/g20/jjellio/src/github/Trilinos/packages/common/auxiliarySoftware/SuiteSparse/src/COLAMD/Include -I/g/g20/jjellio/src/github/Trilinos/packages/common/auxiliarySoftware/SuiteSparse/src/BTF/Include -I/g/g20/jjellio/src/github/Trilinos/packages/common/auxiliarySoftware/SuiteSparse/src/UFconfig -I/tmp/jjellio/build/amd-5.4.3_prgenv-amd_rocm-5.4.3_mpich-8.1.24_pure-amd_hip_amd-inlall-NOfunc-gfx90a_serial_atdm_complex_opt-g_cxx17_static/trilinos/packages/teuchos/core/src -I/g/g20/jjellio/src/github/Trilinos/packages/teuchos/core/src -I/tmp/jjellio/build/amd-5.4.3_prgenv-amd_rocm-5.4.3_mpich-8.1.24_pure-amd_hip_amd-inlall-NOfunc-gfx90a_serial_atdm_complex_opt-g_cxx17_static/trilinos/packages/kokkos/core/src -I/g/g20/jjellio/src/github/Trilinos/packages/kokkos/core/src -I/tmp/jjellio/build/amd-5.4.3_prgenv-amd_rocm-5.4.3_mpich-8.1.24_pure-amd_hip_amd-inlall-NOfunc-gfx90a_serial_atdm_complex_opt-g_cxx17_static/trilinos/packages/kokkos -I/g/g20/jjellio/src/github/Trilinos/packages/kokkos/core/src/../../tpls/desul/include -I/g/g20/jjellio/src/github/Trilinos/packages/teuchos/parser/src -I/g/g20/jjellio/src/github/Trilinos/packages/teuchos/parameterlist/src -I/g/g20/jjellio/src/github/Trilinos/packages/teuchos/comm/src -I/g/g20/jjellio/src/github/Trilinos/packages/teuchos/numerics/src -I/g/g20/jjellio/src/github/Trilinos/packages/teuchos/remainder/src -I/tmp/jjellio/build/amd-5.4.3_prgenv-amd_rocm-5.4.3_mpich-8.1.24_pure-amd_hip_amd-inlall-NOfunc-gfx90a_serial_atdm_complex_opt-g_cxx17_static/trilinos/packages/teuchos/remainder/src -I/tmp/jjellio/build/amd-5.4.3_prgenv-amd_rocm-5.4.3_mpich-8.1.24_pure-amd_hip_amd-inlall-NOfunc-gfx90a_serial_atdm_complex_opt-g_cxx17_static/trilinos/packages/teuchos/kokkoscompat/src -I/g/g20/jjellio/src/github/Trilinos/packages/teuchos/kokkoscompat/src -I/tmp/jjellio/build/amd-5.4.3_prgenv-amd_rocm-5.4.3_mpich-8.1.24_pure-amd_hip_amd-inlall-NOfunc-gfx90a_serial_atdm_complex_opt-g_cxx17_static/trilinos/packages/teuchos/kokkoscomm/src -I/g/g20/jjellio/src/github/Trilinos/packages/teuchos/kokkoscomm/src -I/g/g20/jjellio/src/github/Trilinos/packages/tpetra/tsqr/src -I/tmp/jjellio/build/amd-5.4.3_prgenv-amd_rocm-5.4.3_mpich-8.1.24_pure-amd_hip_amd-inlall-NOfunc-gfx90a_serial_atdm_complex_opt-g_cxx17_static/trilinos/packages/tpetra/tsqr/src -I/tmp/jjellio/build/amd-5.4.3_prgenv-amd_rocm-5.4.3_mpich-8.1.24_pure-amd_hip_amd-inlall-NOfunc-gfx90a_serial_atdm_complex_opt-g_cxx17_static/trilinos/packages/kokkos-kernels/common -I/g/g20/jjellio/src/github/Trilinos/packages/kokkos-kernels/common -I/tmp/jjellio/build/amd-5.4.3_prgenv-amd_rocm-5.4.3_mpich-8.1.24_pure-amd_hip_amd-inlall-NOfunc-gfx90a_serial_atdm_complex_opt-g_cxx17_static/trilinos/packages/kokkos-kernels/batched -I/g/g20/jjellio/src/github/Trilinos/packages/kokkos-kernels/batched -I/tmp/jjellio/build/amd-5.4.3_prgenv-amd_rocm-5.4.3_mpich-8.1.24_pure-amd_hip_amd-inlall-NOfunc-gfx90a_serial_atdm_complex_opt-g_cxx17_static/trilinos/packages/kokkos-kernels/blas -I/g/g20/jjellio/src/github/Trilinos/packages/kokkos-kernels/blas -I/tmp/jjellio/build/amd-5.4.3_prgenv-amd_rocm-5.4.3_mpich-8.1.24_pure-amd_hip_amd-inlall-NOfunc-gfx90a_serial_atdm_complex_opt-g_cxx17_static/trilinos/packages/kokkos-kernels/graph -I/g/g20/jjellio/src/github/Trilinos/packages/kokkos-kernels/graph -I/tmp/jjellio/build/amd-5.4.3_prgenv-amd_rocm-5.4.3_mpich-8.1.24_pure-amd_hip_amd-inlall-NOfunc-gfx90a_serial_atdm_complex_opt-g_cxx17_static/trilinos/packages/kokkos-kernels -I/g/g20/jjellio/src/github/Trilinos/packages/kokkos-kernels/common/src -I/g/g20/jjellio/src/github/Trilinos/packages/kokkos-kernels/common/src/impl -I/g/g20/jjellio/src/github/Trilinos/packages/kokkos-kernels/common/src/tpls -I/g/g20/jjellio/src/github/Trilinos/packages/kokkos-kernels/common/unit_test -I/g/g20/jjellio/src/github/Trilinos/packages/kokkos-kernels/batched/dense/src -I/g/g20/jjellio/src/github/Trilinos/packages/kokkos-kernels/batched/dense/impl -I/g/g20/jjellio/src/github/Trilinos/packages/kokkos-kernels/batched/dense/unit_test -I/g/g20/jjellio/src/github/Trilinos/packages/kokkos-kernels/batched/sparse/src -I/g/g20/jjellio/src/github/Trilinos/packages/kokkos-kernels/batched/sparse/impl -I/g/g20/jjellio/src/github/Trilinos/packages/kokkos-kernels/batched/sparse/unit_test -I/g/g20/jjellio/src/github/Trilinos/packages/kokkos-kernels/blas/src -I/g/g20/jjellio/src/github/Trilinos/packages/kokkos-kernels/blas/impl -I/g/g20/jjellio/src/github/Trilinos/packages/kokkos-kernels/blas/eti -I/tmp/jjellio/build/amd-5.4.3_prgenv-amd_rocm-5.4.3_mpich-8.1.24_pure-amd_hip_amd-inlall-NOfunc-gfx90a_serial_atdm_complex_opt-g_cxx17_static/trilinos/packages/kokkos-kernels/blas/eti -I/g/g20/jjellio/src/github/Trilinos/packages/kokkos-kernels/blas/tpls -I/g/g20/jjellio/src/github/Trilinos/packages/kokkos-kernels/graph/src -I/g/g20/jjellio/src/github/Trilinos/packages/kokkos-kernels/graph/impl -I/g/g20/jjellio/src/github/Trilinos/packages/kokkos-kernels/graph/eti -I/tmp/jjellio/build/amd-5.4.3_prgenv-amd_rocm-5.4.3_mpich-8.1.24_pure-amd_hip_amd-inlall-NOfunc-gfx90a_serial_atdm_complex_opt-g_cxx17_static/trilinos/packages/kokkos-kernels/graph/eti -I/g/g20/jjellio/src/github/Trilinos/packages/kokkos-kernels/sparse/src -I/g/g20/jjellio/src/github/Trilinos/packages/kokkos-kernels/sparse/impl -I/g/g20/jjellio/src/github/Trilinos/packages/kokkos-kernels/sparse/eti -I/tmp/jjellio/build/amd-5.4.3_prgenv-amd_rocm-5.4.3_mpich-8.1.24_pure-amd_hip_amd-inlall-NOfunc-gfx90a_serial_atdm_complex_opt-g_cxx17_static/trilinos/packages/kokkos-kernels/sparse/eti -I/g/g20/jjellio/src/github/Trilinos/packages/kokkos-kernels/sparse/tpls -I/tmp/jjellio/build/amd-5.4.3_prgenv-amd_rocm-5.4.3_mpich-8.1.24_pure-amd_hip_amd-inlall-NOfunc-gfx90a_serial_atdm_complex_opt-g_cxx17_static/trilinos/packages/kokkos/containers/src -I/g/g20/jjellio/src/github/Trilinos/packages/kokkos/containers/src -I/tmp/jjellio/build/amd-5.4.3_prgenv-amd_rocm-5.4.3_mpich-8.1.24_pure-amd_hip_amd-inlall-NOfunc-gfx90a_serial_atdm_complex_opt-g_cxx17_static/trilinos/packages/kokkos/algorithms/src -I/g/g20/jjellio/src/github/Trilinos/packages/kokkos/algorithms/src -I/g/g20/jjellio/src/github/Trilinos/packages/tpetra/core/compat -I/tmp/jjellio/build/amd-5.4.3_prgenv-amd_rocm-5.4.3_mpich-8.1.24_pure-amd_hip_amd-inlall-NOfunc-gfx90a_serial_atdm_complex_opt-g_cxx17_static/trilinos/packages/tpetra/core/compat -I/tmp/jjellio/build/amd-5.4.3_prgenv-amd_rocm-5.4.3_mpich-8.1.24_pure-amd_hip_amd-inlall-NOfunc-gfx90a_serial_atdm_complex_opt-g_cxx17_static/trilinos/packages/epetra/src -I/g/g20/jjellio/src/github/Trilinos/packages/epetra/src -I/g/g20/jjellio/src/github/Trilinos/packages/tpetra/core/src -I/g/g20/jjellio/src/github/Trilinos/packages/tpetra/core/src/kokkos_refactor -I/tmp/jjellio/build/amd-5.4.3_prgenv-amd_rocm-5.4.3_mpich-8.1.24_pure-amd_hip_amd-inlall-NOfunc-gfx90a_serial_atdm_complex_opt-g_cxx17_static/trilinos/packages/tpetra/core/src -I/g/g20/jjellio/src/github/Trilinos/packages/tpetra/core/inout -I/tmp/jjellio/build/amd-5.4.3_prgenv-amd_rocm-5.4.3_mpich-8.1.24_pure-amd_hip_amd-inlall-NOfunc-gfx90a_serial_atdm_complex_opt-g_cxx17_static/trilinos/packages/tpetra/core/inout -I/g/g20/jjellio/src/github/Trilinos/packages/tpetra/core/ext -I/tmp/jjellio/build/amd-5.4.3_prgenv-amd_rocm-5.4.3_mpich-8.1.24_pure-amd_hip_amd-inlall-NOfunc-gfx90a_serial_atdm_complex_opt-g_cxx17_static/trilinos/packages/tpetra/core/ext -I/tmp/jjellio/build/amd-5.4.3_prgenv-amd_rocm-5.4.3_mpich-8.1.24_pure-amd_hip_amd-inlall-NOfunc-gfx90a_serial_atdm_complex_opt-g_cxx17_static/trilinos/packages/common/auxiliarySoftware/SuiteSparse/src -I/g/g20/jjellio/src/github/Trilinos/packages/common/auxiliarySoftware/SuiteSparse/src/CAMD/Include -I/g/g20/jjellio/src/github/Trilinos/packages/common/auxiliarySoftware/SuiteSparse/src/CCOLAMD/Include -I/g/g20/jjellio/src/github/Trilinos/packages/common/auxiliarySoftware/SuiteSparse/src/KLU/Include -I/tmp/jjellio/build/amd-5.4.3_prgenv-amd_rocm-5.4.3_mpich-8.1.24_pure-amd_hip_amd-inlall-NOfunc-gfx90a_serial_atdm_complex_opt-g_cxx17_static/trilinos/packages/kokkos/simd/src -I/g/g20/jjellio/src/github/Trilinos/packages/kokkos/simd/src -I/tmp/jjellio/build/amd-5.4.3_prgenv-amd_rocm-5.4.3_mpich-8.1.24_pure-amd_hip_amd-inlall-NOfunc-gfx90a_serial_atdm_complex_opt-g_cxx17_static/trilinos/packages/epetraext/src -I/g/g20/jjellio/src/github/Trilinos/packages/epetraext/src -I/g/g20/jjellio/src/github/Trilinos/packages/epetraext/src/transform -I/g/g20/jjellio/src/github/Trilinos/packages/epetraext/src/inout -I/g/g20/jjellio/src/github/Trilinos/packages/epetraext/src/coloring -I/g/g20/jjellio/src/github/Trilinos/packages/epetraext/src/model_evaluator -I/g/g20/jjellio/src/github/Trilinos/packages/epetraext/src/block -I/g/g20/jjellio/src/github/Trilinos/packages/epetraext/src/restrict -I/tmp/jjellio/build/amd-5.4.3_prgenv-amd_rocm-5.4.3_mpich-8.1.24_pure-amd_hip_amd-inlall-NOfunc-gfx90a_serial_atdm_complex_opt-g_cxx17_static/trilinos/packages/triutils/src -I/g/g20/jjellio/src/github/Trilinos/packages/triutils/src -I/tmp/jjellio/build/amd-5.4.3_prgenv-amd_rocm-5.4.3_mpich-8.1.24_pure-amd_hip_amd-inlall-NOfunc-gfx90a_serial_atdm_complex_opt-g_cxx17_static/trilinos/packages/shylu/shylu_node/tacho/src -I/g/g20/jjellio/src/github/Trilinos/packages/shylu/shylu_node/tacho/src -I/g/g20/jjellio/src/github/Trilinos/packages/shylu/shylu_node/tacho/src/impl -isystem /p/lustre1/jjellio/spack/install/cray-rhel8-zen3/rocmcc-5.4.3/boost-1.79.0-cy2pon7szje2rpvr33ugc4weoy7cwtwz/include -isystem /opt/cray/pe/libsci/23.02.1.1/AMD/4.0/x86_64/include -isystem /p/lustre1/jjellio/spack/install/cray-rhel8-zen3/rocmcc-5.4.3/metis-5.1.0-63cajq5mqokl5jgjmakg5wnpewwbylds/include -isystem /p/lustre1/jjellio/spack/install/cray-rhel8-zen3/rocmcc-5.4.3/hdf5-1.10.7-e6cwnqznwy36nrn6s5tqq5p4tfw3tbfw/include -isystem /p/lustre1/jjellio/spack/install/cray-rhel8-zen3/rocmcc-5.4.3/parmetis-4.0.3-hs5iqxpe7o2gsvoiqmqx3vsldwkejg4f/include -x hip -mllvm -amdgpu-early-inline-all=true -mllvm -amdgpu-function-calls=false -I/opt/cray/pe/mpich/8.1.24/ofi/amd/5.0/include -I/opt/rocm-5.4.3/include -g -march=znver3 -mtune=znver3 -fno-gpu-rdc -x hip --rocm-path=/opt/rocm-5.4.3 --offload-arch=gfx90a  -O3 -DNDEBUG -std=c++17 -MD -MT packages/amesos2/src/CMakeFiles/amesos2.dir/Amesos2_Lapack.cpp.o -MF packages/amesos2/src/CMakeFiles/amesos2.dir/Amesos2_Lapack.cpp.o.d -o packages/amesos2/src/CMakeFiles/amesos2.dir/Amesos2_Lapack.cpp.o -c /g/g20/jjellio/src/github/Trilinos/packages/amesos2/src/Amesos2_Lapack.cpp
In file included from /g/g20/jjellio/src/github/Trilinos/packages/amesos2/src/Amesos2_Lapack.cpp:49:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/amesos2/src/Amesos2_Lapack_decl.hpp:61:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/amesos2/src/Amesos2_SolverTraits.hpp:59:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/amesos2/src/Amesos2_MatrixAdapter.hpp:47:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/amesos2/src/Amesos2_MatrixAdapter_decl.hpp:55:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/tpetra/core/src/Tpetra_ConfigDefs.hpp:43:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/tpetra/core/src/Tpetra_Details_DefaultTypes.hpp:45:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/tpetra/core/compat/Tpetra_KokkosCompat_ClassicNodeAPI_Wrapper.hpp:4:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/kokkos/core/src/Kokkos_Core.hpp:66:
/g/g20/jjellio/src/github/Trilinos/packages/kokkos/core/src/Kokkos_CopyViews.hpp:284:13: error: reference to __host__ function 'operator complex' in __host__ __device__ function
    a(i0) = static_cast<value_type>(b(i0));
            ^
/g/g20/jjellio/src/github/Trilinos/packages/kokkos/core/src/HIP/Kokkos_HIP_Parallel_Range.hpp:52:5: note: called by 'exec_range<void>'
    m_functor(i);
    ^
/g/g20/jjellio/src/github/Trilinos/packages/kokkos/core/src/HIP/Kokkos_HIP_Parallel_Range.hpp:73:22: note: called by 'operator()'
      this->template exec_range<WorkTag>(iwork);
                     ^
/g/g20/jjellio/src/github/Trilinos/packages/kokkos/core/src/HIP/Kokkos_HIP_KernelLaunch.hpp:80:3: note: called by 'hip_parallel_launch_local_memory<Kokkos::Impl::ParallelFor<Kokkos::Impl::ViewCopy<Kokkos::View<std::complex<double> *, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::HIP, Kokkos::AnonymousSpace>, Kokkos::MemoryTraits<0>>, Kokkos::View<const Kokkos::complex<double> *, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::HIP, Kokkos::AnonymousSpace>, Kokkos::MemoryTraits<0>>, Kokkos::LayoutRight, Kokkos::HIP, 1, long>, Kokkos::RangePolicy<Kokkos::HIP, Kokkos::IndexType<long>>, Kokkos::HIP>, 1024U, 1U>'
  driver();
  ^
/g/g20/jjellio/src/github/Trilinos/packages/kokkos/core/src/Kokkos_Complex.hpp:98:3: note: 'operator complex' declared here
  operator std::complex<RealType>() const noexcept {
  ^
In file included from /g/g20/jjellio/src/github/Trilinos/packages/amesos2/src/Amesos2_Lapack.cpp:49:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/amesos2/src/Amesos2_Lapack_decl.hpp:61:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/amesos2/src/Amesos2_SolverTraits.hpp:59:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/amesos2/src/Amesos2_MatrixAdapter.hpp:47:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/amesos2/src/Amesos2_MatrixAdapter_decl.hpp:55:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/tpetra/core/src/Tpetra_ConfigDefs.hpp:43:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/tpetra/core/src/Tpetra_Details_DefaultTypes.hpp:45:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/tpetra/core/compat/Tpetra_KokkosCompat_ClassicNodeAPI_Wrapper.hpp:4:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/kokkos/core/src/Kokkos_Core.hpp:66:
/g/g20/jjellio/src/github/Trilinos/packages/kokkos/core/src/Kokkos_CopyViews.hpp:284:37: error: reference to __host__ function 'operator complex' in __host__ __device__ function
    a(i0) = static_cast<value_type>(b(i0));
                                    ^
/g/g20/jjellio/src/github/Trilinos/packages/kokkos/core/src/Kokkos_Complex.hpp:98:3: note: 'operator complex' declared here
  operator std::complex<RealType>() const noexcept {
  ^
In file included from /g/g20/jjellio/src/github/Trilinos/packages/amesos2/src/Amesos2_Lapack.cpp:49:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/amesos2/src/Amesos2_Lapack_decl.hpp:61:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/amesos2/src/Amesos2_SolverTraits.hpp:59:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/amesos2/src/Amesos2_MatrixAdapter.hpp:47:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/amesos2/src/Amesos2_MatrixAdapter_decl.hpp:55:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/tpetra/core/src/Tpetra_ConfigDefs.hpp:43:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/tpetra/core/src/Tpetra_Details_DefaultTypes.hpp:45:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/tpetra/core/compat/Tpetra_KokkosCompat_ClassicNodeAPI_Wrapper.hpp:4:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/kokkos/core/src/Kokkos_Core.hpp:66:
/g/g20/jjellio/src/github/Trilinos/packages/kokkos/core/src/Kokkos_CopyViews.hpp:284:13: error: reference to __host__ function 'operator complex' in __host__ __device__ function
    a(i0) = static_cast<value_type>(b(i0));
            ^
/g/g20/jjellio/src/github/Trilinos/packages/kokkos/core/src/HIP/Kokkos_HIP_Parallel_Range.hpp:52:5: note: called by 'exec_range<void>'
    m_functor(i);
    ^
/g/g20/jjellio/src/github/Trilinos/packages/kokkos/core/src/HIP/Kokkos_HIP_Parallel_Range.hpp:73:22: note: called by 'operator()'
      this->template exec_range<WorkTag>(iwork);
                     ^
/g/g20/jjellio/src/github/Trilinos/packages/kokkos/core/src/HIP/Kokkos_HIP_KernelLaunch.hpp:80:3: note: called by 'hip_parallel_launch_local_memory<Kokkos::Impl::ParallelFor<Kokkos::Impl::ViewCopy<Kokkos::View<std::complex<double> *, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::HIP, Kokkos::AnonymousSpace>, Kokkos::MemoryTraits<0>>, Kokkos::View<const Kokkos::complex<double> *, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::HIP, Kokkos::AnonymousSpace>, Kokkos::MemoryTraits<0>>, Kokkos::LayoutLeft, Kokkos::HIP, 1, long>, Kokkos::RangePolicy<Kokkos::HIP, Kokkos::IndexType<long>>, Kokkos::HIP>, 1024U, 1U>'
  driver();
  ^
/g/g20/jjellio/src/github/Trilinos/packages/kokkos/core/src/Kokkos_Complex.hpp:98:3: note: 'operator complex' declared here
  operator std::complex<RealType>() const noexcept {
  ^
In file included from /g/g20/jjellio/src/github/Trilinos/packages/amesos2/src/Amesos2_Lapack.cpp:49:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/amesos2/src/Amesos2_Lapack_decl.hpp:61:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/amesos2/src/Amesos2_SolverTraits.hpp:59:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/amesos2/src/Amesos2_MatrixAdapter.hpp:47:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/amesos2/src/Amesos2_MatrixAdapter_decl.hpp:55:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/tpetra/core/src/Tpetra_ConfigDefs.hpp:43:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/tpetra/core/src/Tpetra_Details_DefaultTypes.hpp:45:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/tpetra/core/compat/Tpetra_KokkosCompat_ClassicNodeAPI_Wrapper.hpp:4:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/kokkos/core/src/Kokkos_Core.hpp:66:
/g/g20/jjellio/src/github/Trilinos/packages/kokkos/core/src/Kokkos_CopyViews.hpp:284:37: error: reference to __host__ function 'operator complex' in __host__ __device__ function
    a(i0) = static_cast<value_type>(b(i0));
                                    ^
/g/g20/jjellio/src/github/Trilinos/packages/kokkos/core/src/Kokkos_Complex.hpp:98:3: note: 'operator complex' declared here
  operator std::complex<RealType>() const noexcept {
  ^
In file included from /g/g20/jjellio/src/github/Trilinos/packages/amesos2/src/Amesos2_Lapack.cpp:49:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/amesos2/src/Amesos2_Lapack_decl.hpp:61:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/amesos2/src/Amesos2_SolverTraits.hpp:59:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/amesos2/src/Amesos2_MatrixAdapter.hpp:47:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/amesos2/src/Amesos2_MatrixAdapter_decl.hpp:55:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/tpetra/core/src/Tpetra_ConfigDefs.hpp:43:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/tpetra/core/src/Tpetra_Details_DefaultTypes.hpp:45:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/tpetra/core/compat/Tpetra_KokkosCompat_ClassicNodeAPI_Wrapper.hpp:4:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/kokkos/core/src/Kokkos_Core.hpp:66:
/g/g20/jjellio/src/github/Trilinos/packages/kokkos/core/src/Kokkos_CopyViews.hpp:284:13: error: reference to __host__ function 'operator complex' in __host__ __device__ function
    a(i0) = static_cast<value_type>(b(i0));
            ^
/g/g20/jjellio/src/github/Trilinos/packages/kokkos/core/src/HIP/Kokkos_HIP_Parallel_Range.hpp:52:5: note: called by 'exec_range<void>'
    m_functor(i);
    ^
/g/g20/jjellio/src/github/Trilinos/packages/kokkos/core/src/HIP/Kokkos_HIP_Parallel_Range.hpp:73:22: note: called by 'operator()'
      this->template exec_range<WorkTag>(iwork);
                     ^
/g/g20/jjellio/src/github/Trilinos/packages/kokkos/core/src/HIP/Kokkos_HIP_KernelLaunch.hpp:80:3: note: called by 'hip_parallel_launch_local_memory<Kokkos::Impl::ParallelFor<Kokkos::Impl::ViewCopy<Kokkos::View<std::complex<double> *, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::HIP, Kokkos::AnonymousSpace>, Kokkos::MemoryTraits<0>>, Kokkos::View<const Kokkos::complex<double> *, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::HIP, Kokkos::AnonymousSpace>, Kokkos::MemoryTraits<0>>, Kokkos::LayoutRight, Kokkos::HIP, 1, int>, Kokkos::RangePolicy<Kokkos::HIP, Kokkos::IndexType<int>>, Kokkos::HIP>, 1024U, 1U>'
  driver();
  ^
/g/g20/jjellio/src/github/Trilinos/packages/kokkos/core/src/Kokkos_Complex.hpp:98:3: note: 'operator complex' declared here
  operator std::complex<RealType>() const noexcept {
  ^
In file included from /g/g20/jjellio/src/github/Trilinos/packages/amesos2/src/Amesos2_Lapack.cpp:49:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/amesos2/src/Amesos2_Lapack_decl.hpp:61:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/amesos2/src/Amesos2_SolverTraits.hpp:59:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/amesos2/src/Amesos2_MatrixAdapter.hpp:47:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/amesos2/src/Amesos2_MatrixAdapter_decl.hpp:55:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/tpetra/core/src/Tpetra_ConfigDefs.hpp:43:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/tpetra/core/src/Tpetra_Details_DefaultTypes.hpp:45:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/tpetra/core/compat/Tpetra_KokkosCompat_ClassicNodeAPI_Wrapper.hpp:4:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/kokkos/core/src/Kokkos_Core.hpp:66:
/g/g20/jjellio/src/github/Trilinos/packages/kokkos/core/src/Kokkos_CopyViews.hpp:284:37: error: reference to __host__ function 'operator complex' in __host__ __device__ function
    a(i0) = static_cast<value_type>(b(i0));
                                    ^
/g/g20/jjellio/src/github/Trilinos/packages/kokkos/core/src/Kokkos_Complex.hpp:98:3: note: 'operator complex' declared here
  operator std::complex<RealType>() const noexcept {
  ^
In file included from /g/g20/jjellio/src/github/Trilinos/packages/amesos2/src/Amesos2_Lapack.cpp:49:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/amesos2/src/Amesos2_Lapack_decl.hpp:61:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/amesos2/src/Amesos2_SolverTraits.hpp:59:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/amesos2/src/Amesos2_MatrixAdapter.hpp:47:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/amesos2/src/Amesos2_MatrixAdapter_decl.hpp:55:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/tpetra/core/src/Tpetra_ConfigDefs.hpp:43:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/tpetra/core/src/Tpetra_Details_DefaultTypes.hpp:45:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/tpetra/core/compat/Tpetra_KokkosCompat_ClassicNodeAPI_Wrapper.hpp:4:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/kokkos/core/src/Kokkos_Core.hpp:66:
/g/g20/jjellio/src/github/Trilinos/packages/kokkos/core/src/Kokkos_CopyViews.hpp:284:13: error: reference to __host__ function 'operator complex' in __host__ __device__ function
    a(i0) = static_cast<value_type>(b(i0));
            ^
/g/g20/jjellio/src/github/Trilinos/packages/kokkos/core/src/HIP/Kokkos_HIP_Parallel_Range.hpp:52:5: note: called by 'exec_range<void>'
    m_functor(i);
    ^
/g/g20/jjellio/src/github/Trilinos/packages/kokkos/core/src/HIP/Kokkos_HIP_Parallel_Range.hpp:73:22: note: called by 'operator()'
      this->template exec_range<WorkTag>(iwork);
                     ^
/g/g20/jjellio/src/github/Trilinos/packages/kokkos/core/src/HIP/Kokkos_HIP_KernelLaunch.hpp:80:3: note: called by 'hip_parallel_launch_local_memory<Kokkos::Impl::ParallelFor<Kokkos::Impl::ViewCopy<Kokkos::View<std::complex<double> *, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::HIP, Kokkos::AnonymousSpace>, Kokkos::MemoryTraits<0>>, Kokkos::View<const Kokkos::complex<double> *, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::HIP, Kokkos::AnonymousSpace>, Kokkos::MemoryTraits<0>>, Kokkos::LayoutLeft, Kokkos::HIP, 1, int>, Kokkos::RangePolicy<Kokkos::HIP, Kokkos::IndexType<int>>, Kokkos::HIP>, 1024U, 1U>'
  driver();
  ^
/g/g20/jjellio/src/github/Trilinos/packages/kokkos/core/src/Kokkos_Complex.hpp:98:3: note: 'operator complex' declared here
  operator std::complex<RealType>() const noexcept {
  ^
In file included from /g/g20/jjellio/src/github/Trilinos/packages/amesos2/src/Amesos2_Lapack.cpp:49:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/amesos2/src/Amesos2_Lapack_decl.hpp:61:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/amesos2/src/Amesos2_SolverTraits.hpp:59:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/amesos2/src/Amesos2_MatrixAdapter.hpp:47:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/amesos2/src/Amesos2_MatrixAdapter_decl.hpp:55:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/tpetra/core/src/Tpetra_ConfigDefs.hpp:43:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/tpetra/core/src/Tpetra_Details_DefaultTypes.hpp:45:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/tpetra/core/compat/Tpetra_KokkosCompat_ClassicNodeAPI_Wrapper.hpp:4:
In file included from /g/g20/jjellio/src/github/Trilinos/packages/kokkos/core/src/Kokkos_Core.hpp:66:
/g/g20/jjellio/src/github/Trilinos/packages/kokkos/core/src/Kokkos_CopyViews.hpp:284:37: error: reference to __host__ function 'operator complex' in __host__ __device__ function
    a(i0) = static_cast<value_type>(b(i0));
                                    ^
/g/g20/jjellio/src/github/Trilinos/packages/kokkos/core/src/Kokkos_Complex.hpp:98:3: note: 'operator complex' declared here
  operator std::complex<RealType>() const noexcept {
  ^
8 errors generated when compiling for gfx90a.
```
</details>


<details>
<summary> my_configure.sh </summary>

```
cmake \
 "-GNinja" \
 "-DCMAKE_BUILD_TYPE:STRING=Release" \
 "-DTrilinos_ENABLE_TrilinosBuildStats=OFF" \
 "-DTrilinos_ENABLE_BUILD_STATS=OFF" \
 "-DTrilinosBuildStats_ENABLE_TESTS=OFF" \
 "-DBUILD_SHARED_LIBS:BOOL=OFF" \
 "-DTrilinos_ENABLE_Teuchos=ON" \
 "-DTrilinos_ENABLE_Percept=OFF" \
 "-DTrilinos_ENABLE_ALL_PACKAGES=ON" \
 "-DKokkos_ARCH_ZEN3:BOOL=ON" \
 "-DKokkos_ENABLE_SERIAL:BOOL=ON" \
 "-DTpetra_INST_SERIAL:BOOL=ON" \
 "-DKokkos_ENABLE_OPENMP:BOOL=OFF" \
 "-DTpetra_INST_OPENMP:BOOL=OFF" \
 "-DTrilinos_ENABLE_OpenMP:BOOL=OFF" \
 "-DKokkos_ENABLE_HIP:BOOL=ON" \
 "-DTpetra_INST_HIP:BOOL=ON" \
 "-DKokkos_ARCH_VEGA90A:BOOL=ON" \
 "-DCMAKE_EXE_LINKER_FLAGS=-x none --hip-link  " \
 "-DCMAKE_SHARED_LINKER_FLAGS=-x none --hip-link  " \
 "-DFC_FN_UNDERSCORE=UNDER" \
 "-DCMAKE_CXX_COMPILER=/opt/rocm-5.4.3/llvm/bin/amdclang++" \
 "-DCMAKE_C_COMPILER=/opt/rocm-5.4.3/llvm/bin/amdclang" \
 "-DCMAKE_Fortran_COMPILER=/opt/rocm-5.4.3/llvm/bin/amdflang" \
 "-DCMAKE_CXX_FLAGS=-x hip -mllvm -amdgpu-early-inline-all=true -mllvm -amdgpu-function-calls=false -I/opt/cray/pe/mpich/8.1.24/ofi/amd/5.0/include -I/opt/rocm-5.4.3/include -g " \
 "-DCMAKE_Fortran_FLAGS=-I/opt/cray/pe/mpich/8.1.24/ofi/amd/5.0/include -I/opt/rocm-5.4.3/include " \
 "-DCMAKE_C_FLAGS=-I/opt/cray/pe/mpich/8.1.24/ofi/amd/5.0/include -I/opt/rocm-5.4.3/include " \
 "-DTrilinos_EXTRA_LINK_FLAGS=-L/opt/rocm-5.4.3/lib -Wl,-rpath,/opt/rocm-5.4.3/lib -lamdhip64 -L/opt/cray/pe/mpich/8.1.24/ofi/amd/5.0/lib -Wl,-rpath,/opt/cray/pe/mpich/8.1.24/ofi/amd/5.0/lib -lmpi_amd -L/opt/cray/pe/mpich/8.1.24/gtl/lib -Wl,-rpath,/opt/cray/pe/mpich/8.1.24/gtl/lib -lmpi_gtl_hsa -L/opt/rocm-5.4.3/llvm/lib/clang/15.0.0/lib/linux/ -lclang_rt.builtins-x86_64 " \
 "-DCMAKE_CXX_STANDARD=17" \
 "-DTrilinos_ENABLE_Fortran:BOOL=OFF" \
 "-DAztecOO_C_FLAGS=-Wno-implicit-function-declaration" \
 "-DTrilinos_ENABLE_TrilinosATDMConfigTests:BOOL=OFF" \
 "-DTrilinos_ENABLE_TrilinosFrameworkTests=OFF" \
 "-DTrilinos_ENABLE_MiniTensor=OFF" \
 "-DTrilinos_ENABLE_Isorropia=OFF" \
 "-DTrilinos_ENABLE_KokkosExample=OFF" \
 "-DTrilinos_ENABLE_Domi=OFF" \
 "-DTrilinos_ENABLE_Pliris=OFF" \
 "-DTrilinos_ENABLE_Komplex=OFF" \
 "-DTrilinos_ENABLE_FEI=OFF" \
 "-DTrilinos_ENABLE_TriKota=OFF" \
 "-DTrilinos_ENABLE_Compadre=OFF" \
 "-DTrilinos_ENABLE_STKClassic=OFF" \
 "-DTrilinos_ENABLE_STKSearchUtil=OFF" \
 "-DTrilinos_ENABLE_STKUnit_tests=OFF" \
 "-DTrilinos_ENABLE_STKDoc_tests=OFF" \
 "-DTrilinos_ENABLE_STKExp=OFF" \
 "-DTrilinos_ENABLE_Moertel=OFF" \
 "-DTrilinos_ENABLE_Stokhos=OFF" \
 "-DTrilinos_ENABLE_MOOCHO=OFF" \
 "-DTrilinos_ENABLE_PyTrilinos=OFF" \
 "-DTrilinos_ENABLE_TrilinosCouplings=OFF" \
 "-DTrilinos_ENABLE_Pike=OFF" \
 "-DTrilinos_ENABLE_Krino=OFF" \
 "-DShyLU_DD_ENABLE_BDDC=OFF" \
 "-DTrilinos_ENABLE_PyTrilinos=OFF" \
 "-DTrilinos_ENABLE_Tempus=OFF" \
 "-DKrino_ENABLE_TESTS=OFF" \
 "-DROL_ENABLE_TESTS=OFF" \
 "-DROL_ENABLE_EXAMPLES=OFF" \
 "-DKOKKOSKERNELS_TPL_BLAS_RETURN_COMPLEX=OFF" \
 "-DTacho_ENABLE_INT_INT:BOOL=ON" \
 "-DTrilinos_ENABLE_TESTS:BOOL=OFF" \
 "-DPanzerMiniEM_ENABLE_EXAMPLES=ON" \
 "-DPanzerMiniEM_ENABLE_TESTS=ON" \
 "-DZoltan_ENABLE_TESTS=OFF" \
 "-DTrilinos_ENABLE_COMPLEX_DOUBLE:BOOL=ON" \
 "-DTPL_ENABLE_BinUtils:BOOL=OFF" \
 "-DBinUtils_INCLUDE_DIRS=/p/lustre1/jjellio/spack/install/cray-rhel8-zen3/cce-14.0.0/binutils-2.38-qtwpwyu4a23d5534lgbjyw6zwmorh44b/include" \
 "-DBinUtils_LIBRARY_DIRS=/p/lustre1/jjellio/spack/install/cray-rhel8-zen3/cce-14.0.0/binutils-2.38-qtwpwyu4a23d5534lgbjyw6zwmorh44b/lib" \
 "-DTPL_ENABLE_BLAS:BOOL=ON" \
 "-DTPL_ENABLE_LAPACK:BOOL=ON" \
 "-DBLAS_LIBRARY_NAMES=sci_amd" \
 "-DBLAS_INCLUDE_DIRS=/opt/cray/pe/libsci/23.02.1.1/AMD/4.0/x86_64/include" \
 "-DBLAS_LIBRARY_DIRS=/opt/cray/pe/libsci/23.02.1.1/AMD/4.0/x86_64/lib" \
 "-DLAPACK_LIBRARY_NAMES=sci_amd" \
 "-DLAPACK_INCLUDE_DIRS=/opt/cray/pe/libsci/23.02.1.1/AMD/4.0/x86_64/include" \
 "-DLAPACK_LIBRARY_DIRS=/opt/cray/pe/libsci/23.02.1.1/AMD/4.0/x86_64/lib" \
 "-DTPL_ENABLE_Boost:BOOL=ON" \
 "-DTPL_ENABLE_BoostLib:BOOL=ON" \
 "-DBoost_INCLUDE_DIRS=/p/lustre1/jjellio/spack/install/cray-rhel8-zen3/rocmcc-5.4.3/boost-1.79.0-cy2pon7szje2rpvr33ugc4weoy7cwtwz/include" \
 "-DBoost_LIBRARY_DIRS=/p/lustre1/jjellio/spack/install/cray-rhel8-zen3/rocmcc-5.4.3/boost-1.79.0-cy2pon7szje2rpvr33ugc4weoy7cwtwz/lib" \
 "-DBoostLib_INCLUDE_DIRS=/p/lustre1/jjellio/spack/install/cray-rhel8-zen3/rocmcc-5.4.3/boost-1.79.0-cy2pon7szje2rpvr33ugc4weoy7cwtwz/include" \
 "-DBoostLib_LIBRARY_DIRS=/p/lustre1/jjellio/spack/install/cray-rhel8-zen3/rocmcc-5.4.3/boost-1.79.0-cy2pon7szje2rpvr33ugc4weoy7cwtwz/lib" \
 "-DTPL_ENABLE_METIS:BOOL=ON" \
 "-DTPL_ENABLE_ParMETIS:BOOL=ON" \
 "-DTPL_METIS_INCLUDE_DIRS=/p/lustre1/jjellio/spack/install/cray-rhel8-zen3/rocmcc-5.4.3/metis-5.1.0-63cajq5mqokl5jgjmakg5wnpewwbylds/include" \
 "-DTPL_METIS_LIBRARY_DIRS=/p/lustre1/jjellio/spack/install/cray-rhel8-zen3/rocmcc-5.4.3/metis-5.1.0-63cajq5mqokl5jgjmakg5wnpewwbylds/lib" \
 "-DTPL_ParMETIS_INCLUDE_DIRS=/p/lustre1/jjellio/spack/install/cray-rhel8-zen3/rocmcc-5.4.3/parmetis-4.0.3-hs5iqxpe7o2gsvoiqmqx3vsldwkejg4f/include;/p/lustre1/jjellio/spack/install/cray-rhel8-zen3/rocmcc-5.4.3/metis-5.1.0-63cajq5mqokl5jgjmakg5wnpewwbylds/include" \
 "-DTPL_ParMETIS_LIBRARY_DIRS=/p/lustre1/jjellio/spack/install/cray-rhel8-zen3/rocmcc-5.4.3/parmetis-4.0.3-hs5iqxpe7o2gsvoiqmqx3vsldwkejg4f/lib;/p/lustre1/jjellio/spack/install/cray-rhel8-zen3/rocmcc-5.4.3/metis-5.1.0-63cajq5mqokl5jgjmakg5wnpewwbylds/lib" \
 "-DTPL_ENABLE_CGNS:BOOL=ON" \
 "-DCGNS_INCLUDE_DIRS=/p/lustre1/jjellio/spack/install/cray-rhel8-zen3/rocmcc-5.4.3/cgns-4.3.0-6i7szcbr7rr57avhmifsapfpwaz2egqw/include" \
 "-DCGNS_LIBRARY_DIRS=/p/lustre1/jjellio/spack/install/cray-rhel8-zen3/rocmcc-5.4.3/cgns-4.3.0-6i7szcbr7rr57avhmifsapfpwaz2egqw/lib" \
 "-DTPL_ENABLE_HDF5:BOOL=ON" \
 "-DHDF5_LIBRARY_DIRS=/p/lustre1/jjellio/spack/install/cray-rhel8-zen3/rocmcc-5.4.3/hdf5-1.10.7-e6cwnqznwy36nrn6s5tqq5p4tfw3tbfw/lib;/p/lustre1/jjellio/spack/install/cray-rhel8-zen3/rocmcc-5.4.3/zlib-1.2.12-zebpynaf4xxoupoytpmrnmfwcm6jv4xd/lib" \
 "-DHDF5_INCLUDE_DIRS=/p/lustre1/jjellio/spack/install/cray-rhel8-zen3/rocmcc-5.4.3/hdf5-1.10.7-e6cwnqznwy36nrn6s5tqq5p4tfw3tbfw/include" \
 "-DHDF5_LIBRARY_NAMES=hdf5_hl;hdf5;z;dl" \
 "-DTPL_ENABLE_Netcdf:BOOL=ON" \
 "-DNetcdf_LIBRARY_DIRS=/p/lustre1/jjellio/spack/install/cray-rhel8-zen3/rocmcc-5.4.3/zlib-1.2.12-zebpynaf4xxoupoytpmrnmfwcm6jv4xd/lib;/p/lustre1/jjellio/spack/install/cray-rhel8-zen3/rocmcc-5.4.3/boost-1.79.0-cy2pon7szje2rpvr33ugc4weoy7cwtwz/lib;/p/lustre1/jjellio/spack/install/cray-rhel8-zen3/rocmcc-5.4.3/netcdf-c-4.8.1-hmkfol7k4ve2pvj2rmk3szs742snvv2w/lib;/p/lustre1/jjellio/spack/install/cray-rhel8-zen3/rocmcc-5.4.3/parallel-netcdf-1.12.3-utp5t2ujb7kszxk4ktlvilwlg6acrgox/lib;/p/lustre1/jjellio/spack/install/cray-rhel8-zen3/rocmcc-5.4.3/hdf5-1.10.7-e6cwnqznwy36nrn6s5tqq5p4tfw3tbfw/lib;/p/lustre1/jjellio/spack/install/cray-rhel8-zen3/rocmcc-5.4.3/libzip-1.2.0-iaxw654gssjadssjrv46lproy5ahmmap/lib" \
 "-DNetcdf_LIBRARY_NAMES=netcdf;pnetcdf;z;zip;hdf5_hl;hdf5" \
 "-DNetcdf_INCLUDE_DIRS=/p/lustre1/jjellio/spack/install/cray-rhel8-zen3/rocmcc-5.4.3/netcdf-c-4.8.1-hmkfol7k4ve2pvj2rmk3szs742snvv2w/include;/p/lustre1/jjellio/spack/install/cray-rhel8-zen3/rocmcc-5.4.3/parallel-netcdf-1.12.3-utp5t2ujb7kszxk4ktlvilwlg6acrgox/include" \
 "-DTPL_ENABLE_SuperLUDist:BOOL=OFF" \
 "-DSuperLUDist_INCLUDE_DIRS=/p/lustre1/jjellio/spack/install/cray-rhel8-zen3/rocmcc-5.4.3/superlu-dist-6.4.0-ekgyze4beemtgp5zmyzefzrbnh4mhkpf/include" \
 "-DSuperLUDist_LIBRARY_DIRS=/p/lustre1/jjellio/spack/install/cray-rhel8-zen3/rocmcc-5.4.3/superlu-dist-6.4.0-ekgyze4beemtgp5zmyzefzrbnh4mhkpf/lib" \
 "-DTPL_ENABLE_Matio=OFF" \
 "-DTPL_ENABLE_X11=OFF" \
 "-DTPL_ENABLE_Gtest=OFF" \
 "-DTrilinos_ENABLE_Gtest=OFF" \
 "-DTrilinos_MAKE_INSTALL_GROUP=elcapnda" \
 "-DTrilinos_MAKE_INSTALL_GROUP_READABLE:BOOL=TRUE" \
 "-DTrilinos_MAKE_INSTALL_GROUP_WRITABLE:BOOL=FALSE" \
 "-DTrilinos_MAKE_INSTALL_WORLD_READABLE:BOOL=TRUE" \
 "-DTPL_ENABLE_MPI:BOOL=ON" \
 "-DMPI_USE_COMPILER_WRAPPERS=OFF" \
 "-DMPI_EXEC_NUMPROCS_FLAG=mini;run;-o;exit-timeout=10s;-N;1;-o;mpibind=off;-o;cpu-affinity=per-task;-o;gpu-affinity=per-task;-c6;--gpus-per-task=1;-n" \
 "-DMPI_EXEC=flux" \
 "-DTPL_ENABLE_DLlib:BOOL=ON" \
 "-DDLlib_INCLUDE_DIRS=/opt/rocm-5.4.3/include" \
 "-DDLlib_LIBRARY_DIRS=/opt/rocm-5.4.3/lib" \
 "-DDLlib_LIBRARY_NAMES=dl;hipsolver;rocsolver;hipblas;rocblas;hipsparse;rocsparse;amd_comgr;hsa-runtime64;amdhip64" \
 "-DCMAKE_INSTALL_PREFIX=/p/lustre1/jjellio/installs/rzvernal/amd-5.4.3_prgenv-amd_rocm-5.4.3_mpich-8.1.24_pure-amd_hip_amd-inlall-NOfunc-gfx90a_serial_atdm_complex_opt-g_cxx17_static" \
/g/g20/jjellio/src/github/Trilinos
```
</details>

The build is probably hard to reproduce at SNL, but should be reproducable on Crusher.

--- a/packages/amesos2/src/Amesos2_Lapack_decl.hpp
+++ b/packages/amesos2/src/Amesos2_Lapack_decl.hpp
@@ -96,12 +96,15 @@ namespace Amesos2 {
     typedef typename super_type::global_ordinal_type              global_ordinal_type;
     typedef typename super_type::global_size_type                    global_size_type;
 
+    // Version of scalar type that can be used in Kokkos device code
+    typedef typename MatrixTraits<Matrix>::impl_scalar_type          impl_scalar_type;
+
     typedef typename Teuchos::ScalarTraits<scalar_type>::magnitudeType magnitude_type;
 
     typedef Kokkos::DefaultHostExecutionSpace              HostExecSpaceType;
     typedef Kokkos::View<int*, HostExecSpaceType>          host_size_type_array;
     typedef Kokkos::View<int*, HostExecSpaceType>          host_ordinal_type_array;
-    typedef Kokkos::View<scalar_type*, HostExecSpaceType> host_value_type_array;
+    typedef Kokkos::View<impl_scalar_type*, HostExecSpaceType>  host_value_type_array;
 
     /// \name Constructor/Destructor methods
     //@{ 

--- a/packages/amesos2/src/Amesos2_MatrixTraits.hpp
+++ b/packages/amesos2/src/Amesos2_MatrixTraits.hpp
@@ -85,6 +85,7 @@ namespace Amesos2 {
     typedef Node node_t;
 
     typedef Tpetra::RowMatrix<Scalar, LocalOrdinal, GlobalOrdinal, Node>  matrix_type;
+    typedef typename matrix_type::impl_scalar_type                    impl_scalar_type;
 
     typedef typename matrix_type::nonconst_global_inds_host_view_type global_host_idx_type;
     typedef typename matrix_type::nonconst_values_host_view_type      global_host_val_type;
@@ -107,6 +108,7 @@ namespace Amesos2 {
     typedef Node node_t;
 
     typedef Tpetra::CrsMatrix<Scalar, LocalOrdinal, GlobalOrdinal, Node>  matrix_type;
+    typedef typename matrix_type::impl_scalar_type                    impl_scalar_type;
 
     typedef typename matrix_type::nonconst_global_inds_host_view_type global_host_idx_type;
     typedef typename matrix_type::nonconst_values_host_view_type      global_host_val_type;
@@ -127,6 +129,7 @@ namespace Amesos2 {
     typedef LocalOrdinal global_size_t;
 
     typedef KokkosSparse::CrsMatrix<Scalar, LocalOrdinal, DeviceType>  matrix_type;
+    typedef Scalar impl_scalar_type;
     typedef Tpetra::Map<>::node_type node_t;
 
     typedef typename matrix_type::HostMirror::index_type  global_host_idx_type;
@@ -140,6 +143,7 @@ namespace Amesos2 {
   template <>
   struct MatrixTraits<Epetra_RowMatrix> {
     typedef double scalar_t;
+    typedef double impl_scalar_type;
     typedef int local_ordinal_t;
     typedef Tpetra::Map<>::global_ordinal_type global_ordinal_t;
     typedef Tpetra::Map<>::node_type node_t;
@@ -159,6 +163,7 @@ namespace Amesos2 {
   template <>
   struct MatrixTraits<Epetra_CrsMatrix> {
     typedef double scalar_t;
+    typedef double impl_scalar_type;
     typedef int local_ordinal_t;
     typedef Tpetra::Map<>::global_ordinal_type global_ordinal_t;
     typedef Tpetra::Map<>::node_type node_t;
@@ -188,6 +193,7 @@ namespace Amesos2 {
   template <>
   struct MatrixTraits<Epetra_VbrMatrix> {
     typedef double scalar_t;
+    typedef double impl_scalar_type;
     typedef int local_ordinal_t;
     typedef Tpetra::Map<>::global_ordinal_type global_ordinal_t;
     typedef Tpetra::Map<>::node_type node_t;

